src/api/custom-resolvers/send-email-password-reset-link.ts

```diff
@@ -0,0 +1,66 @@
+import { Context } from '.keystone/types'; // eslint-disable-line
+import crypto from 'crypto';
+import { ACCOUNT, FIELD_IDS } from '../constants';
+import getAccountByField from '../helpers/get-account-by-field';
+import sendEmail from '../emails';
+import { AccountSendEmailPasswordResetLinkVariables, Account, SuccessResponse } from '../types';
+
+const {
+  ENCRYPTION: {
+    STRING_TYPE,
+    PBKDF2: { ITERATIONS, DIGEST_ALGORITHM },
+    PASSWORD: {
+      PBKDF2: { KEY_LENGTH },
+    },
+  },
+} = ACCOUNT;
+
+/**
+ * sendEmailPasswordResetLink
+ * - TODO
+ * @param {Object} GraphQL root variables
+ * @param {Object} GraphQL variables for the SendEmailPasswordResetLink mutation
+ * @param {Object} KeystoneJS context API
+ * @returns {Object} Object with success flag
+ */
+const sendEmailPasswordResetLink = async (root: any, variables: AccountSendEmailPasswordResetLinkVariables, context: Context): Promise<SuccessResponse> => {
+  try {
+    console.info('Sending password reset email');
+
+    const { email } = variables;
+
+    // Get the account the email is associated with.
+    const exporter = await getAccountByField(context, FIELD_IDS.ACCOUNT.EMAIL, email);
+
+    if (!exporter) {
+      console.info('Unable to send password reset email - no account found');
+
+      return { success: false };
+    }
+
+    const passwordResetHash = crypto.pbkdf2Sync(email, exporter.salt, ITERATIONS, KEY_LENGTH, DIGEST_ALGORITHM).toString(STRING_TYPE);
```

Comment:
@abhi-markan Keen to get any feedback from you on this please - this is where we generate a password reset hash with 5mins expiry and save it in the DB. Uses same approach/encryption as DTFS.

Next PRs will be about handling/consuming the hash/reset link sent via email.

Reply:
In the interests of progressing the sprint and unblocking PRs, going to walk through password reset as a whole in one session and can make changes after this if required